pkg/syncer/resource_syncer.go

```diff
@@ -137,16 +155,20 @@ type ResourceSyncerConfig struct {
 
 	// ResyncPeriod if non-zero, the period at which resources will be re-synced regardless if anything changed. Default is 0.
 	ResyncPeriod time.Duration
+
+	// MetricOpts used to pass name and help text to resource syncer Gauge
+	MetricOpts *prometheus.GaugeOpts
 }
 
 type resourceSyncer struct {
-	workQueue workqueue.Interface
-	informer  cache.Controller
-	store     cache.Store
-	config    ResourceSyncerConfig
-	deleted   sync.Map
-	created   sync.Map
-	stopped   chan struct{}
+	workQueue   workqueue.Interface
+	informer    cache.Controller
+	store       cache.Store
+	config      ResourceSyncerConfig
+	deleted     sync.Map
+	created     sync.Map
+	stopped     chan struct{}
+	metricGauge *prometheus.GaugeVec
```

Comment:
Instead of storing the type in the member name, and indicating only that it’s a metric, let’s describe what it’s used for:

```suggestion
	syncCounter *prometheus.GaugeVec
```